### PR TITLE
Added CMake and CI configuration for inflate compilation options

### PR DESCRIPTION
**Pull request by nmoinvaz**
This PR adds CMake and CI configuration for `INFLATE_STRICT` and `INFLATE_ALLOW_INVALID_DISTANCE_TOOFAR_ARRR`.

### CODE REVIEW

**Comment by Dead2** [comment]
Same comments are valid here as in #525, these options should not be displayed by default.

**Comment by nmoinvaz** [comment]
When #525 is merged, I can revise this PR along the same lines as whatever is decided there.

**Comment by nmoinvaz** [comment]
Rebased and I changed these to be advanced options.

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,9 +103,11 @@ elseif(BASEARCH_X86_FOUND)
     option(WITH_SSE4 "Build with SSE4" ON)
     option(WITH_PCLMULQDQ "Build with PCLMULQDQ" ON)
 endif()
+option(WITH_INFLATE_STRICT "Build with strict inflate distance checking" OFF)
+option(WITH_INFLATE_ALLOW_INVALID_DIST "Build with zero fill for inflate invalid distances" OFF)
```
Comment (Dead2) [inline CMakeLists.txt]: These should be advanced options (and moved up over the basearch bunch)

Comment (nmoinvaz) [reply CMakeLists.txt]: It is already part of `mark_as_advanced`

Comment (nmoinvaz) [reply CMakeLists.txt]: I have moved those two lines above basearch bunch as you suggested.

**Pull request merged**